### PR TITLE
Add syntax highlight for code blocks

### PR DESCRIPTION
**Pull request by nkilm**
## Additions
- Used [rehype-pretty-code](https://rehype-pretty-code.netlify.app/) for [build-time](https://rehype-pretty-code.netlify.app/#:~:text=There%27s%20no%20runtime%20or%20bundle%20size%20cost) syntax highlighting. 
- Theme used - `one-dark-pro`, theme can be changed in [next.config.js](https://github.com/umami-software/website/blob/master/next.config.js).

## Related Issues 
#103 - Syntax Highlighting in docs

## Example
checkout the code snippet here  - [/docs/running-on-vercel](https://website-git-fork-nkilm-syntax-highlight-umami-software.vercel.app/docs/running-on-vercel)

```js
// mentioning programming language is important to highlight appropriate keywords
<script
  async
  defer
  src="/stats/script.js"
  data-website-id="94db1cb1-74f4-4a40-ad6c-962362670409"
></script>
```

image

## References
- [https://rehype-pretty-code.netlify.app/](https://rehype-pretty-code.netlify.app/)
- Supported themes - [shiki-themes](https://github.com/shikijs/shiki/blob/main/docs/themes.md)

### CODE REVIEW

File: content/running-on-vercel.mdx
Comment (uncenter) [inline content/running-on-vercel.mdx:1]: This file shouldn't have any changes.

File: next.config.js [hunk 1]
```diff
@@ -1,8 +1,16 @@
+const rehypePrettyCode = require('rehype-pretty-code');
+
+const rehypePrettyCodeOptions = {
+  // prepacked themes from shiki
+  // Themes list - https://github.com/shikijs/shiki/blob/main/docs/themes.md
+  theme: 'one-dark-pro',
```
Comment (uncenter) [inline next.config.js:6]: Theme looks good! You could probably remove the comment though.

**Review by uncenter — CHANGES_REQUESTED** [review]
Looks solid though.

File: next.config.js (hunk 1 above)
Comment (nkilm) [reply next.config.js:6]: Alright, I'll remove the comments.

File: content/running-on-vercel.mdx
Comment (nkilm) [reply content/running-on-vercel.mdx:1]: Alright, I'll update the code in this branch by merging with main. We do have to mention the language for code snippets to get proper syntax highlight.

Comment (uncenter) [reply content/running-on-vercel.mdx:1]: Yeah I made a PR a little while back to add languages for all of the code blocks so this isn't necessary. Thanks for getting on this so quick!

Comment (nkilm) [reply content/running-on-vercel.mdx:1]: That's great! I'll push the requested changes asap.

**Comment by uncenter** [comment]
Nice work! One last thing. I was taking a look at the code blocks on the website and they look a little dull/grayish with the current theme. I messed around with a bunch of themes and `github-light` really looks nice. 

image
image
image
image

Thoughts? It seems much sharper and stands out more.

**Comment by nkilm** [comment]
Yeah I was about mention that!😅

I'll change the theme to `github-light`.

**Comment by nkilm** [comment]
@uncenter Done✅

File: package.json [hunk 2]
```diff
@@ -40,8 +40,11 @@
     "react-dom": "^18.2.0",
     "react-scroll-parallax": "^3.2.0",
     "react-spring": "^8.0.27",
+    "react-tooltip": "^4.2.21",
```
Comment (uncenter) [inline package.json:43]: Why is `react-tooltip` added?

Comment (uncenter) [reply package.json:43]: I'm guessing it somehow got added in the merge commit but I'm not seeing it on the `master` branch...

Comment (nkilm) [reply package.json:43]: Yeah I didn't notice this part, I was having some merge conflicts this got added there. I'll remove it.

**Comment by nkilm** [comment]
@uncenter I've removed that package.

**Comment by uncenter** [comment]
> @uncenter I've removed that package.

Did you run the uninstall command or just remove it from the package.json? Looks like you had a build failure as soon as you did that.

**Comment by nkilm** [comment]
> > @uncenter I've removed that package.
> 
> Did you run the uninstall command or just remove it from the package.json? Looks like you had a build failure as soon as you did that.

I didn't remove it using `yarn uninstall`. I did it manually. I'll fix it now, also are there any other changes which are to be made?

**Comment by uncenter** [comment]
I think that's it. If the build succeeds + no unnecessary files are changed, it should be good to go.

**Comment by nkilm** [comment]
Alright

**Comment by nkilm** [comment]
@uncenter I've fixed the issue now, build is successful.

**Review by uncenter — APPROVED** [review]

**Comment by mikecao** [comment]
We've made major changes to the website, so instead of dealing with merge conflicts, I implemented your solution. It's live on the website now. Closing this PR.

**Comment by uncenter** [comment]
Great! New design + syntax highlighting looking good.

**Pull request closed without merging**